luci_grpc_interface/interface/src/interface.cpp

```diff
@@ -376,30 +378,47 @@ void Interface::processRearIrData()
 void Interface::sendJsCallback(const luci_messages::msg::LuciJoystick::SharedPtr msg)
 {
     // Send the remote JS values over gRPC
-    spdlog::debug("Received js val: {} {}", msg->forward_back, msg->left_right);
+    auto inputSource = static_cast<InputSource>(msg->input_source);
+    spdlog::debug("Received js val: {} {} {}", msg->forward_back, msg->left_right,
+                  msg->input_source);
 
-    this->luciInterface->sendJS(msg->forward_back + 100, msg->left_right + 100);
+    this->luciInterface->sendJS(msg->forward_back + 100, msg->left_right + 100, inputSource);
 }
 
-void Interface::switchLuciModeCallback(const luci_messages::msg::LuciDriveMode::SharedPtr msg)
+void Interface::setRemoteInputSource()
 {
-    // Send a mode switch over gRPC
-    spdlog::info("Received mode switch request");
-    switch (msg->mode)
+    int checker = this->luciInterface->setInputSource(InputSource::Remote);
+    std::chrono::milliseconds waitTime(100);
+    if (checker == 0)
     {
-    case luci_messages::msg::LuciDriveMode::USER:
-        this->luciInterface->activateUserMode();
-        break;
-    case luci_messages::msg::LuciDriveMode::ENGAGED:
-        this->luciInterface->activateEngagedMode();
-        break;
-    case luci_messages::msg::LuciDriveMode::AUTO:
-        this->luciInterface->activateAutoMode();
-        break;
-    default:
-        spdlog::error("NOT A VALID MODE SELECTION");
-        break;
+        this->luciInterface->sendJS(100, 100, InputSource::Remote);
+        while (true)
```

Comment:
I think we will want to document in the SDK docs somewhere that this can potentially hang forever or add an error path that times this out if it takes too long. I believe the only expected way this can hang at the moment is if the chair is in ramp assist, but I'm not sure that will be true forever.